static/src/javascripts-legacy/projects/common/modules/experiments/tests/tailor-survey.js

```diff
@@ -0,0 +1,152 @@
+        this.audienceCriteria = 'All users';
+        this.dataLinkNames = 'Tailor survey';
+        this.idealOutcome = '';
+
+        this.canRun = function () {
+            return !(config.page.isAdvertisementFeature) &&
+                config.page.contentType === 'Article'
+        };
+
+        function callTailor(bwid) {
+            var endpoint = 'https://tailor.guardianapis.com/suggestions?browserId=' + bwid;
+            return fetchJson(endpoint, {
+                type: 'json',
+                method: 'get'
+            });
+        }
+
+        function renderQuickSurvey() {
+            var bwid = cookies.get('bwid');
+            var hasSeenTheSurveyAlready = cookies.get('GU_TAILOR_SURVEY_1') || false;
+
+            if (bwid && !hasSeenTheSurveyAlready) {
+                return callTailor(bwid).then(function (response) {
+                    return response.userDataForClient.regular;
+                }).then(function(isRegular) {
+                    if (isRegular) {
+                        cookies.add('GU_TAILOR_SURVEY_1', 1, 100); // do not show this survey to the user for the next 100 days
+
+                        return fastdomPromise.write(function () {
+                            var article = document.getElementsByClassName('content__article-body')[0];
+                            var insertionPoint = article.getElementsByTagName('p')[1];
+                            var surveyDiv = document.createElement('div');
+                            surveyDiv.innerHTML = quickSurvey;
+                            article.insertBefore(surveyDiv, insertionPoint);
+                        });
+                    }
+                });
+            }
+        }
+
+        function disableRadioButtons(buttonClassName) {
+            var radioButtons = document.getElementsByClassName(buttonClassName);
+            for (var i = 0; i < radioButtons.length; i++) {
+                radioButtons[i].disabled=true;
+            }
+        }
+
+        function surveyFadeOut() {
+            var surveyContent = document.getElementsByClassName('impressions-survey__content');
+            fastdom.write(function () {
+                surveyContent[0].classList.add('js-impressions-survey__fadeout');
+            });
+        }
+
+        function thankyouFadeIn() {
+            var surveyThanks = document.getElementsByClassName('impressions-survey__thanks');
+            fastdom.write(function () {
+                surveyThanks[0].classList.add('js-impressions-survey__fadein');
+            });
+        }
+
+        function handleSurveyResponse() {
+            var surveyQuestions = document.getElementsByClassName('fi-survey__button');
+
+            forEach(surveyQuestions, function(question) {
+                bean.on(question, 'click', function (event) {
+                    
+                    var answer = event.target.attributes.getNamedItem("data-link-name").value;
```

Comment:
What if the `data-link-name` attribute is deleted in the html, but not in the JS. In that case this would throw an exception. Best just to check for it before using it.